presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/MapColumnValidator.java

```diff
@@ -0,0 +1,140 @@
+            keysChecksum = new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(sortKeys));
+        }
+        else {
+            keysChecksum = new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(mapKeys));
+        }
+
+        Expression valuesChecksum;
+        FunctionCall mapValues = new FunctionCall(QualifiedName.of("map_values"), ImmutableList.of(column.getIdentifier()));
+        if (valueType.isOrderable()) {
+            FunctionCall sortValues = new FunctionCall(QualifiedName.of("array_sort"), ImmutableList.of(mapValues));
+            valuesChecksum = new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(sortValues));
+        }
+        else {
+            valuesChecksum = new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(mapValues));
+        }
+
+        Expression mapCardinalitySum = new CoalesceExpression(
+                new FunctionCall(
+                        QualifiedName.of("sum"),
+                        ImmutableList.of(new FunctionCall(QualifiedName.of("cardinality"), ImmutableList.of(column.getIdentifier())))),
+                new LongLiteral("0"));
+
+        return ImmutableList.of(
+                new SingleColumn(checksum, Optional.of(delimitedIdentifier(getChecksumColumnAlias(column)))),
+                new SingleColumn(keysChecksum, Optional.of(delimitedIdentifier(getKeysChecksumColumnAlias(column)))),
+                new SingleColumn(valuesChecksum, Optional.of(delimitedIdentifier(getValuesChecksumColumnAlias(column)))),
+                new SingleColumn(mapCardinalitySum, Optional.of(delimitedIdentifier(getCardinalitySumColumnAlias(column)))));
+    }
+
+    @Override
+    public ColumnMatchResult validate(Column column, ChecksumResult controlResult, ChecksumResult testResult)
+    {
+        String checksumColumnAlias = getChecksumColumnAlias(column);
+        Object controlChecksum = controlResult.getChecksum(checksumColumnAlias);
+        Object testChecksum = testResult.getChecksum(checksumColumnAlias);
+
+        String keysChecksumColumnAlias = getKeysChecksumColumnAlias(column);
+        Object controlKeysChecksum = controlResult.getChecksum(keysChecksumColumnAlias);
+        Object testKeysChecksum = testResult.getChecksum(keysChecksumColumnAlias);
+
+        String valuesChecksumColumnAlias = getValuesChecksumColumnAlias(column);
+        Object controlValuesChecksum = controlResult.getChecksum(valuesChecksumColumnAlias);
+        Object testValuesChecksum = testResult.getChecksum(valuesChecksumColumnAlias);
+
+        String cardinalitySumColumnAlias = getCardinalitySumColumnAlias(column);
+        Object controlCardinalitySum = controlResult.getChecksum(cardinalitySumColumnAlias);
+        Object testCardinalitySum = testResult.getChecksum(cardinalitySumColumnAlias);
+
+        return new ColumnMatchResult(
+                Objects.equals(controlChecksum, testChecksum) && Objects.equals(controlCardinalitySum, testCardinalitySum),
```

Comment:
Also make sure key checksum matches and and value checksum matches.
nit: one condition per line for long line:
```Java
Object.equals(...)
    && Object.equals(...)
    && Object.equals(...)
    && Object.equals(...)
```